src/geomData.js

```diff
@@ -604,11 +619,30 @@ vgl.sourceData = function(arg) {
     var i;
 
     //m_data = m_data.concat(data); //no, slow on Safari
+    /* If we will are given a Float32Array and don't have any other data, use
+     * it directly */
+    if (!m_data.length && data.length &&
+        Object.prototype.toString.call(data) === "[object Float32Array]") {
+      m_data = data;
+      return;
+    }
+    /* If our internal array is immutable and we will need to change it, create
```

Comment:
I am sorry, yes I guess we never changed completely to // because of jsdoc issues. Don't worry about those then.